scripts/tests/run_tv_casting_test.py

```diff
@@ -0,0 +1,228 @@
+
+                # Initialize variables to store the information of interest
+                valid_vendor_id = False
+                valid_product_id = False
+                valid_device_type = False
+
+                valid_vendor_id_str = ''
+                valid_product_id_str = ''
+                valid_device_type_str = ''
+
+            if continue_parsing:
+
+                # Check if the Vendor ID, Product ID, and Device Type match the expected values
+                if "Vendor ID:" in line:
+
+                    # If the value of the Vendor ID does not match the expected value, then
+                    # handle the discovery failure.
+                    valid_vendor_id = validate_value(VENDOR_ID, line, "Vendor ID")
+
+                    if not valid_vendor_id:
+                        handle_discovery_failure()
+                    else:
+                        line = line.rstrip('\n')
+                        valid_vendor_id_str = line
+
+                elif "Product ID:" in line:
+
+                    # If the value of Product ID does not match the expected value, then
+                    # handle the discovery failure.
+                    valid_product_id = validate_value(PRODUCT_ID, line, "Product ID")
+
+                    if not valid_product_id:
+                        handle_discovery_failure()
+                    else:
+                        line = line.rstrip('\n')
+                        valid_product_id_str = line
+
+                elif "Device Type:" in line:
+
+                    # If the value of Device Type does not match the expected value, then
+                    # handle the discovery failure.
+                    valid_device_type = validate_value(DEVICE_TYPE, line, "Device Type")
+
+                    if not valid_device_type:
+                        handle_discovery_failure()
+                    else:
+                        line = line.rstrip('\n')
+                        valid_device_type_str = line
+
+                        # At this point, all values of interest are valid, so we stop parsing.
+                        continue_parsing = False
+
+                # We only print the discovered commissioner that has valid vendor id, product id,
+                # and device type. Remove the log files once done.
+                if valid_vendor_id and valid_product_id and valid_device_type:
+                    print(valid_discovered_commissioner_str)
+                    print(valid_vendor_id_str)
+                    print(valid_product_id_str)
+                    print(valid_device_type_str)
+                    print('Discovery success!')
+
+                    remove_log_file(LINUX_TV_CASTING_APP_LOGS)
+                    remove_log_file(LINUX_TV_APP_LOGS)
+
+                    break
+
+        # Tear down the processes.
+        tv_app_process.terminate()
+        tv_app_process.wait()
+
+        tv_casting_app_process.terminate()
+        tv_casting_app_process.wait()
```

Comment:
Can you make this a `def tear_down`, that you just call from there?

Reply:
Added `def tear_down` in the latest commit.